backend/src/org/commcare/resources/model/ResourceTable.java

```diff
@@ -141,29 +142,33 @@ public void addResource(Resource resource, ResourceInstaller initializer,
     }
 
     public void addResource(Resource resource, int status) {
-        // only add resource if they don't already exist
         if (storage.getIDsForValue(Resource.META_INDEX_RESOURCE_ID,
```

Comment:
If you're going to delete that comment, maybe factor this check out into a function called like resourceDoesNotExist()